src/app/pages/toolbar.component.ts

```diff
@@ -39,16 +40,23 @@ class ToolbarComponent extends HTMLElement {
 		this.modes = this.querySelector('app-modes');
 		this.settings = this.querySelector('app-settings');
 
-		this.localStorageService.getItem('modeOfUse').then((result: any) => {
-			this.json = result;
+		this.filesService.getModesOfUse().then((result: any) => {
+			this.defaultJson = result;
 
-			if (!result) {
-				this.filesService.getModesOfUse().then((result: any) => {
-					this.localStorageService.setItem('modeOfUse', result);
-				});
-			}
+			this.localStorageService.getItem('modeOfUse').then((result: any) => {
+				this.json = result;
+				if (!result) {
```

Comment:
Ça m’interpelle ici : si on a pas de `result`, que vaut `this.json` ? Est-ce qu’on ne devrait pas définir `this.json` dans un `else`, parès la vérification de `result` ?

Je n’en suis pas certain, mais l’ordre me surprend en tout cas.